tests/test_devices.py

```diff
@@ -255,6 +255,152 @@ def test_device_inventory_update_success(client, mocker):
     assert device._inventory == inventory
 
 
+class test_inputs:
+    def test_property_populated(client, mocker):
+        """Should check if the device property are correctly populated"""
+        device = AlarmDevice(client)
+        mocker.spy(device._connection, "query")
+        device.connect("username", "password")
+        inputs = {
+            0: "Entryway Sensor",
+            1: "Outdoor Sensor 1",
+            2: "Outdoor Sensor 2",
+        }
+        # Test
+        device.update()
+        device_inputs = {}
+        for input_id, name in device.inputs:
+            device_inputs.update({input_id: name})
+        assert device_inputs == inputs
+
+    def test_all_properties_empty(client, mocker):
+        """Should check store the e-connect System status in the inventory device object."""
+        device = AlarmDevice(client)
+        mocker.spy(device._connection, "query")
+        device.connect("username", "password")
+        # Test
+        device.update()
+        device._inventory = {}
+        device_inputs = list(device.inputs)
+        assert not device_inputs
+
+    def test_input_property_empty(client, mocker):
+        """Should check store the e-connect System status in the inventory device object."""
+        device = AlarmDevice(client)
+        mocker.spy(device._connection, "query")
+        device.connect("username", "password")
+        # Test
+        device.update()
+        device._inventory.update({"inputs": {}})
```

Comment:
It doesn't matter really much, but in this case you can simply:
```python
# Solution 1: you define exactly what _inventory should be
device._inventory = {"inputs": {}}

# Solution 2: you override "inputs" to be sure it's empty
device._inventory["inputs"] = {}

# Solution 3: you use update as you are doing
device._inventory.update({"inputs": {}})
```

Solution 2 and 3 are the same, and using (2) is syntactically better. (1) is probably the best because it ignores what is the rest of `_inventory` as you are only verifying "inputs".

Up to you what to pick!